nohang

```diff
@@ -707,94 +768,149 @@ if 'rate_zram' in config_dict:
         print('Invalid rate_zram value, not float\nExit')
         exit()
     if rate_zram <= 0:
-        print('rate_zram должен быть положительным\nExit')
+        print('rate_zram MUST be > 0\nExit')
         exit()
 else:
     print('rate_zram not in config\nExit')
     exit()
 
 
-if 'mem_min_sigterm' in config_dict:
-    mem_min_sigterm = config_dict['mem_min_sigterm']
-
-    if mem_min_sigterm.endswith('%'):
-        # отбрасываем процент, получаем число
-        mem_min_sigterm_percent = mem_min_sigterm[:-1].strip()
-        # далее флоат тест
-        mem_min_sigterm_percent = string_to_float_convert_test(mem_min_sigterm_percent)
-        if mem_min_sigterm_percent is None:
-            print('Invalid mem_min_sigterm value, not float\nExit')
-            exit()
-        # окончательная валидация
-        if mem_min_sigterm_percent < 0 or mem_min_sigterm_percent > 100:
-            print('mem_min_sigterm, выраженный в процентах, должен быть быть в диапазоне [0; 100]\nExit')
-            exit()
-
-        # mem_min_sigterm_percent это теперь чистое валидное флоат число процентов, можно переводить в кб
-        mem_min_sigterm_kb = mem_min_sigterm_percent / 100 * mem_total
-        mem_min_sigterm_mb = round(mem_min_sigterm_kb / 1024)
-
-    elif mem_min_sigterm.endswith('M'):
-        mem_min_sigterm_mb = string_to_float_convert_test(mem_min_sigterm[:-1].strip())
-        if mem_min_sigterm_mb is None:
-            print('Invalid mem_min_sigterm value, not float\nExit')
-            exit()
-        mem_min_sigterm_kb = mem_min_sigterm_mb * 1024
-        if mem_min_sigterm_kb > mem_total:
-            print('mem_min_sigterm value не должен быть больше MemTotal ({} MiB)\nExit'.format(round(mem_total / 1024)))
-            exit()
-        mem_min_sigterm_percent = mem_min_sigterm_kb / mem_total * 100
-
-    else:
-        print('Конфиг инвалид, для mem_min_sigterm неверно указаны единицы измерения\nExit')
-        exit()
-
-else:
-    print('mem_min_sigterm not in config\nExit')
-    exit()
-
-
-if 'mem_min_sigkill' in config_dict:
-    mem_min_sigkill = config_dict['mem_min_sigkill']
+def calculate_percent(arg_key):
```

Comment:
You can use a single function to read the required value, since, as far as I can see, the algorithms for all values are identical. Python additionally optimizes the function, what should a good impact on code performance.